File: More_R_and_python_code/MapSimple.R

### Happiness

# Libraries (only run once)
if (!requireNamespace("tigris")) install.packages("tigris")
if (!requireNamespace("sf")) install.packages("sf")
if (!requireNamespace("arrow")) install.packages("arrow")
if (!requireNamespace("patchwork")) install.packages("patchwork")
if (!requireNamespace("ggplot2")) install.packages("ggplot2")
if (!requireNamespace("data.table")) install.packages("data.table")

library(tigris)
library(sf)
library(arrow)
library(patchwork)
library(ggplot2)
library(data.table)

# -----------------------------
# Load datasets from Hugging Face
# -----------------------------
state_data <- data.table(read_parquet("https://huggingface.co/datasets/siacus/flourishing/resolve/main/flourishingStateYear.parquet"))
county_data <- data.table(read_parquet("https://huggingface.co/datasets/siacus/flourishing/resolve/main/flourishingCountyYear.parquet"))

# -----------------------------
# Filter for happiness and a specific year
# -----------------------------
vars <- unique(state_data$variable)
vars

var <- "happiness"
yr <- 2012

state_plot_data <- state_data[variable == var & year == yr]
county_plot_data <- county_data[variable == var & year == yr]

# Ensure correct FIPS formatting
state_plot_data[, FIPS := sprintf("%02s", FIPS)]
county_plot_data[, FIPS_county := sprintf("%05s", FIPS_county)]

# -----------------------------
# Load and prep shapefiles
# -----------------------------
# States
states <- states(cb = TRUE, resolution = "20m", year = 2021) %>% st_as_sf()
states <- states[!states$STUSPS %in% c("AK", "HI", "PR"), ]
states$STATEFP <- sprintf("%02s", states$STATEFP)

# Counties
counties <- counties(cb = TRUE, resolution = "20m", year = 2021) %>% st_as_sf()
counties <- counties[!counties$STATEFP %in% c("02", "15", "72"), ]
counties$GEOID <- sprintf("%05s", counties$GEOID)

# -----------------------------
# Merge data with maps
# -----------------------------
map_state <- merge(states, state_plot_data, by.x = "STATEFP", by.y = "FIPS", all.x = TRUE)
map_county <- merge(counties, county_plot_data, by.x = "GEOID", by.y = "FIPS_county", all.x = TRUE)

# -----------------------------
# Create plots
# -----------------------------
p_state <- ggplot(map_state) +
  geom_sf(aes(fill = stat), color = "white") +
  scale_fill_viridis_c(option = "plasma", na.value = "grey90") +
  labs(
    title = paste("Happiness Stat by State -", yr),
    fill = "Stat"
  ) +
  theme_minimal()

p_county <- ggplot(map_county) +
  geom_sf(aes(fill = stat), color = NA) +
  scale_fill_viridis_c(option = "plasma", na.value = "grey90") +
  labs(
    title = paste("Happiness Stat by County -", yr),
    fill = "Stat"
  ) +
  theme_minimal()

# -----------------------------
# Combine and show
# -----------------------------
p_state + p_county

File: More_R_and_python_code/MapSimple.py
import geopandas as gpd
import pandas as pd
import matplotlib.pyplot as plt
from datasets import load_dataset

# -----------------------------
# Load data from Hugging Face
# -----------------------------
state_ds = load_dataset("siacus/flourishing", data_files="flourishingStateYear.parquet")
county_ds = load_dataset("siacus/flourishing", data_files="flourishingCountyYear.parquet")

state_df = state_ds["train"].to_pandas()
county_df = county_ds["train"].to_pandas()

# -----------------------------
# Filter for happiness and specific year
# -----------------------------
var = "happiness"
yr = 2012

state_plot = state_df[(state_df["variable"] == var) & (state_df["year"] == yr)].copy()
county_plot = county_df[(county_df["variable"] == var) & (county_df["year"] == yr)].copy()

state_plot["FIPS"] = state_plot["FIPS"].apply(lambda x: f"{int(x):02d}")
county_plot["FIPS_county"] = county_plot["FIPS_county"].apply(lambda x: f"{int(x):05d}")

# -----------------------------
# Load shapefiles
# -----------------------------
# US State and County boundaries
states = gpd.read_file("https://www2.census.gov/geo/tiger/GENZ2021/shp/cb_2021_us_state_20m.zip")
counties = gpd.read_file("https://www2.census.gov/geo/tiger/GENZ2021/shp/cb_2021_us_county_20m.zip")

# Remove AK, HI, PR
states = states[~states["STUSPS"].isin(["AK", "HI", "PR"])]
counties = counties[~counties["STATEFP"].isin(["02", "15", "72"])]

# -----------------------------
# Merge with flourishing data
# -----------------------------
states["FIPS"] = states["STATEFP"]
state_map = states.merge(state_plot, on="FIPS", how="left")

counties["FIPS_county"] = counties["STATEFP"] + counties["COUNTYFP"]
county_map = counties.merge(county_plot, on="FIPS_county", how="left")

# -----------------------------
# Plot maps side by side
# -----------------------------
fig, axes = plt.subplots(1, 2, figsize=(18, 8))

# State-level map
state_map.plot(
    column="stat",
    cmap="plasma",
    linewidth=0.1,
    ax=axes[0],
    edgecolor="white",
    missing_kwds={"color": "lightgrey"}
)
axes[0].set_title(f"Happiness Stat by State - {yr}")
axes[0].axis("off")

# County-level map
county_map.plot(
    column="stat",
    cmap="plasma",
    linewidth=0,
    ax=axes[1],
    edgecolor="white",
    missing_kwds={"color": "lightgrey"}
)
axes[1].set_title(f"Happiness Stat by County - {yr}")
axes[1].axis("off")

plt.tight_layout()
plt.show()
